libs/zzz/db/src/driveDiskScanner.ts

```diff
@@ -58,15 +58,37 @@ export function DDSToZOD(data: DDS[]): IZenlessObjectDescription {
               typeof subVal === 'string' && subVal.endsWith('%')
             const [keyStr, value] = keyUp.split('+')
             const upgrades = value ? parseInt(value) + 1 : 1
-            const key = allDiscSubStatKeys.find(
+            let key = allDiscSubStatKeys.find(
               (k) =>
                 isPercentStat(k) === isSubPercent &&
                 statKeyTextMap[k] === keyStr
             )
+            // Address issue scanning atk% as atk
+            if (
+              key === mainStatKey &&
+              key === 'atk' &&
+              typeof subVal === 'number' &&
+              subVal === 57
+            )
+              key = 'atk_'
+
             if (!key || !upgrades) return null
             return { key, upgrades }
           })
           .filter(notEmpty)
+
+        if (substats.length > 4) {
```

Comment:
Can't you just read the main stat as both 1 and 2-lines and see which one is correct? (e.g. which one actually parses out to a proper mainstat)

Reply:
it's substats, not main stat, and the scanner "auto corrects"(wrongly) the value so I can't extract the upgrades from the value